Source/ProgressionSystemRuntime/Public/ModuleStructures.h

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Yevhenii Selivanov
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Engine/DataTable.h"
+#include "Bomber.h"
+#include "Structures/PlayerTag.h"
+#include "ModuleStructures.generated.h"
+
+
+USTRUCT(BlueprintType)
+struct FProgressionRowData : public FTableRowBase
+{
+	GENERATED_BODY()
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category="C++")
+	ELevelType Map;
```

Comment:
in case of struct just assign null?

Reply:
most of all structs have own empty static instance like `FPlayerTag Character = FPlayerTag::None`. If it's your own struct, create such instance for yourself. If it's missing for Engine's struct, just leave uninitialized

Reply:
done

Reply:
Please inline initialize all `FProgressionRowData` properties in header (like in UPSCSaveGame class and others), for it create static const Empty property. _See example in FPlayerTag::None (and it's definition in .cpp)_

Reply:
FPlayerTag actually has None as default, which replicated the FGameplayTag::EmptyTag. 
I can't find where the EmptyTag initiated with "empty" values. For me looks like it's not initiated 

```
/** An empty Gameplay Tag */
	static GAMEPLAYTAGS_API const FGameplayTag EmptyTag;
```

in the header file (GameplayTagContainer.h). The constructor is empty as well: 
```
/** Constructors */
	FGameplayTag()
	{
	}
```

 In the implementation file (GameplayTagContainer.cpp): 

`const FGameplayTag FGameplayTag::EmptyTag;`

am I missing something? 

![image](https://github.com/h4rdmol/ProgressionSystem/assets/5227233/5ea94c53-8d2f-4f16-a0ce-15872d2b3f0d)

![image](https://github.com/h4rdmol/ProgressionSystem/assets/5227233/273efa5f-c648-43c6-adfc-47a9c9e4f6eb)

Reply:
- FProgressionRowData SavedProgressionRowDataInternal = FProgressionRowData::EmptyData;
This uses the copy constructor. EmptyData is pre-initialized, and its values are simply copied to SavedProgressionRowDataInternal. No default constructor is called for SavedProgressionRowDataInternal during this operation.
Uninitialized version:

- FProgressionRowData SavedProgressionRowDataInternal;
Here, SavedProgressionRowDataInternal remains uninitialized until it is explicitly assigned or constructed. This can lead to undefined behavior if the object is used before proper initialization. In terms of construction, there's no initial overhead, but it requires careful management to ensure it's properly initialized before use.

In summary, the EmptyData initialization ensures a known state from the start, using efficient copy construction. The uninitialized version might save initial construction time but requires careful handling to avoid undefined behavior.

What you need to do:
- In .h:
```
static const FProgressionRowData EmptyData;
```
- In .cpp:
```
const FProgressionRowData FProgressionRowData::EmptyData = FProgressionRowData();
```

Reply:
done, review required